gsoc/gsoc-2023.md

```diff
@@ -0,0 +1,159 @@
+
+**Skills Required**
+* HTML/CSS/Javascript familiarity
+* Some notion of what JSON Schema is -- or equivalently some experience using a similar data validation tool including writing SQL Schemas for databases
+* Basic experience with [any of a broad set of languages with JSON Schema implementations](https://json-schema.org/implementations.html#validators) or curiosity to learn enough about them to write a simple program
+* Some Python experience a strong plus
+
+**Mentors**
+[Julian Berman](https://github.com/Julian)
+
+**Expected Difficulty**
+Easy-Medium
+
+**Expected Time Commitment**
+175 Hour
+
+---
+### [#315](https://github.com/json-schema-org/community/issues/315) Extension the official test suite of JSON Schema
+
+**Brief Description**
+The JSON Schema test suite is a collection of reviewed examples which support creating compliant implementations of the JSON Schema specification(s) by verifying their validation result against the expected ones. The existing test suite, which covers validation results (whether a particular piece of data is valid or invalid under a schema) is mature, containing tests across all modern JSON Schema versions. Most recently, JSON Schema has grown to include the notion of annotations, which are pieces of metadata collected whilst processing an instance using a schema. These annotations are used to define various JSON Schema keywords, but also are useful for building JSON Schema tooling beyond simple valid/invalid validation results.
+An initial set of annotation tests has been [added to the suite](https://github.com/json-schema-org/JSON-Schema-Test-Suite/tree/main/output-tests) to begin work on providing tests for this functionality, but it needs extending to fully cover the set of annotation-related functionality in the specification.
+
+**Expected Outcomes**
+Outcomes of this work should include:
+- Carefully read and understand the JSON Schema specifications with a focus on annotations, combing the spec for test cases which should be added, and raising discussion questions in cases where behavior seems unclear in the specification
+- Production of a suite of JSON test cases which fully cover of all JSON Schema keywords which produce annotations, by adding corresponding tests for each one (or each scenario) alongside their expected annotation output for all (3) versions of JSON Schema which contain annotation-related behavior
+- Authoring of a simple annotation collection library, either by re-implementing "throwaway" annotation support in an existing implementation which already has it (e.g. https://github.com/gregsdennis/json-everything) or in a net-new implementation. The purpose of doing so should specifically be to ensure that the suite of tests added "fully cover" the relevant functionality (i.e. that they test all scenarios needed for annotation collection)
```

Comment:
@Julian We discussed this line specifically on the OCWM call yesterday. Not sure what "throwaway" annotation support IS. Gregs implementation, AFAIK, supports full annotation collection and relays that information in the resulting output.

Regardless, to me it feels like a big ask for someone of little expereince to author an annotation collection library. It's basically "implement JSON Schema" to start with, isn't it? If not, I'm not sure what you're suggesting.

Regardless, I think we probably have to push on with the submission as the deadline looms.

Reply:
I see this was discussed https://github.com/json-schema-org/community/issues/315

Reply:
Yeah I mean that was precisely the point of saying "maybe just reimplement in an existing library", because that way when you get stuck you can cheat and re-add the existing bits. But seems like clearly this line was confusing.

Reply:
(it's not strictly necessary of course, but it's pretty much the way to know the new tests are any good, so if the gsoc contributor doesn't do it obviously we can do it after/during ourselves)